plugins/omelasticsearch/omelasticsearch.c

```diff
@@ -1084,14 +1151,14 @@ static rsRetVal
 curlPost(wrkrInstanceData_t *pWrkrData, uchar *message, int msglen, uchar **tpls, int nmsgs)
 {
 	CURLcode code;
-	CURL *curl = pWrkrData->curlHandle;
+	CURL *curl = pWrkrData->curlPostHandle;
 	DEFiRet;
 
 	pWrkrData->reply = NULL;
 	pWrkrData->replyLen = 0;
 
-	if(pWrkrData->pData->dynSrchIdx || pWrkrData->pData->dynSrchType || pWrkrData->pData->dynParent)
-		CHKiRet(setCurlURL(pWrkrData, pWrkrData->pData, tpls));
+	CHKiRet(checkConn(pWrkrData));
```

Comment:
Also, I opted for a very naive approach to health check, where it doesn't store status from one request to the next. 

As an enhancement, we could implement periodic health check (maintain state) and backoff (delay checking after failure), but I didn't think it was necessary now